flocker/control/_config.py

```diff
@@ -295,7 +297,8 @@ def convert_volume(self):
                 u'mountpoint': self._application.volume.mountpoint.path
             }
             dataset = self._application.volume.dataset
-            if dataset.dataset_id is not None:
+            if dataset.dataset_id != dataset_id_from_name(
+                    self._application.name):
```

Comment:
This doesn't seem strictly necessary?  Except it makes some overly specific tests continue to pass instead of breaking them with a change they shouldn't actually be concerned with, which maybe is a sufficient reason to do it...

It probably merits a comment, at least - either saying it's only done to avoid having to do a bunch of boring test maintenance, or else giving another good reason if there is one.